### PR TITLE
rav4 hybrid 2019 port

### PR DESCRIPTION
**Pull request by wocsor**
this time i actually updated the readme. waiting on cabana logs

### CODE REVIEW

File: selfdrive/car/toyota/values.py [hunk 1]
```diff
@@ -178,6 +179,9 @@ def check_ecu_msgs(fingerprint, ecu):
   {
     36: 8, 37: 8, 166: 8, 170: 8, 180: 8, 295: 8, 296: 8, 401: 8, 426: 6, 452: 8, 466: 8, 467: 8, 550: 8, 552: 4, 560: 7, 562: 6, 581: 5, 608: 8, 610: 8, 643: 7, 658: 8, 713: 8, 728: 8, 740: 5, 742: 8, 743: 8, 744: 8, 761: 8, 764: 8, 765: 8, 800: 8, 810: 2, 812: 8, 814: 8, 818: 8, 824: 8, 829: 2, 830: 7, 835: 8, 836: 8, 863: 8, 865: 8, 869: 7, 870: 7, 871: 2, 877: 8, 881: 8, 882: 8, 885: 8, 889: 8, 896: 8, 898: 8, 900: 6, 902: 6, 905: 8, 913: 8, 918: 8, 921: 8, 933: 8, 934: 8, 935: 8, 944: 8, 945: 8, 950: 8, 951: 8, 953: 8, 955: 8, 956: 8, 971: 7, 975: 5, 987: 8, 993: 8, 998: 5, 999: 7, 1000: 8, 1001: 8, 1002: 8, 1017: 8, 1020: 8, 1041: 8, 1042: 8, 1056: 8, 1057: 8, 1059: 1, 1071: 8, 1076: 8, 1077: 8, 1082: 8, 1084: 8, 1085: 8, 1086: 8, 1114: 8, 1132: 8, 1161: 8, 1162: 8, 1163: 8, 1164: 8, 1165: 8, 1166: 8, 1167: 8, 1172: 8, 1228: 8, 1235: 8, 1264: 8, 1279: 8, 1541: 8, 1552: 8, 1553: 8, 1556: 8, 1557: 8, 1568: 8, 1570: 8, 1571: 8, 1572: 8, 1575: 8, 1592: 8, 1594: 8, 1595: 8, 1649: 8, 1696: 8, 1775: 8, 1777: 8, 1779: 8, 1786: 8, 1787: 8, 1788: 8, 1789: 8, 1904: 8, 1912: 8, 1990: 8, 1998: 8
   }],
+  CAR.RAV4H_TSS2: [
```
Comment (rbiasini) [inline selfdrive/car/toyota/values.py]: typo, missing {

File: selfdrive/car/toyota/values.py [hunk 2]
```diff
@@ -178,6 +179,9 @@ def check_ecu_msgs(fingerprint, ecu):
...
+    36: 8, 37: 8, 166: 8, 170: 8, 180: 8, 295:8, 296: 8, 401: 8, 426: 6, 452: 8, 466: 8, 467: 8, 550: 8, 552: 4, 560: 7, 562: 6, 581: 5, 608: 8, 610: 8, 643: 7, 713: 8, 728: 8, 740: 5, 742: 8, 743: 8, 761: 8, 764: 8, 765: 8, 800: 8, 810: 2, 812: 8, 818: 8, 824: 8, 829: 2, 830: 7, 835: 8, 836: 8, 863: 8, 865: 8, 869: 7,870: 7, 871: 2, 877: 8, 881: 8, 882: 8, 885: 8, 889: 8, 896: 8, 898: 8, 900: 6, 902: 6, 905: 8, 921: 8, 933: 8, 934: 8, 935: 8, 944: 8, 945: 8, 950: 8, 951: 8, 953:8, 955: 8, 956: 8, 971: 7, 975: 5, 993: 8, 998: 5, 999: 7, 1000: 8, 1001: 8, 1002: 8, 1014: 8, 1017: 8, 1020:8, 1041: 8, 1042: 8, 1044: 8, 1056: 8, 1057: 8, 1059: 1, 1063: 8, 1071: 8, 1076: 8, 1077: 8, 1084: 8, 1085: 8, 1086: 8, 1114: 8, 1132: 8, 1161: 8, 1162: 8, 1163: 8,1164: 8, 1165: 8, 1166: 8, 1167: 8, 1172: 8, 1235: 8, 1237: 8, 1264: 8, 1279: 8, 1541: 8, 1552: 8, 1553: 8, 1556: 8, 1557: 8, 1568: 8, 1570: 8, 1571: 8, 1572: 8, 1592: 8, 1594: 8, 1595: 8, 1649: 8, 1745: 8, 1775: 8, 1779: 8, 1786: 8, 1787: 8, 1788: 8, 1789: 8, 1904: 8, 1912: 8, 1990: 8, 1998: 8
+  ]},
```
Comment (rbiasini) [inline selfdrive/car/toyota/values.py]: typo, should be }]

**Comment by wocsor** [comment]
it could have been 1 commit if i were more careful... fixed those typos now

**Comment by rbiasini** [comment]
@wocsor do you have a route you can share? I've looked at dongle 76280b332ccbc474 and I've found a larger fingerprint

**Comment by wocsor** [comment]
https://my.comma.ai/cabana/?route=0fb8905d1fa7dbdb%7C2019-07-14--17-26-57&max=27&url=https%3A%2F%2Fchffrprivate-vzn.azureedge.net%2Fchffrprivate3%2Fv2%2F0fb8905d1fa7dbdb%2F187b38b0e5e38c1bdc4cf42c073a773f_2019-07-14--17-26-57

sorry about all the little errors btw

**Comment by rbiasini** [comment]
no worries. Unfortunately this fingerprint conflicts with some of the the Camry Hybrid. It will need to wait until VIN query, which is coming soon.

**Comment by ErichMoraga** [comment]
ETA on that VIN query @rbiasini ???  :-)

**Comment by wocsor** [comment]
funny story on that, over the past week a friend and I did a lot of under the hood work on openpilot and found that there is a working VIN query in car helpers, but if a valid VIN is written to params boardd will change the safety to nooutput. after commenting that out it then only writes the first message per bus in sendcan. we haven't gone any further but we suspect another blocker in boardd is the issue

**Comment by ErichMoraga** [comment]
:crying:

**Comment by illumiN8i** [comment]
Assisted sideplugmike on Discord with getting this working on 2019 Rav4 Hybrid XSE.
Dongle ID 689f254b15632b32

Stock Drive: https://my.comma.ai/cabana/?route=689f254b15632b32%7C2019-08-20--09-38-35&max=11&url=https%3A%2F%2Fchffrprivate-vzn.azureedge.net%2Fchffrprivate3%2Fv2%2F689f254b15632b32%2Fa22611d366b256d72767078c4bb8a25a_2019-08-20--09-38-35

Camry Hybrid fingerprints had to be commented out to resolve conflict.

2019 Rav4 Hybrid XSE Fingerprint: 36: 8, 37: 8, 166: 8, 170: 8, 180: 8, 295: 8, 296: 8, 401: 8, 426: 6, 452: 8, 466: 8, 467: 8, 550: 8, 552: 4, 560: 7, 562: 6, 581: 5, 608: 8, 610: 8, 643: 7, 658: 8, 713: 8, 728: 8, 740: 5, 742: 8, 743: 8, 761: 8, 764: 8, 765: 8, 800: 8, 810: 2, 812: 8, 814: 8, 818: 8, 824: 8, 829: 2, 830: 7, 835: 8, 836: 8, 863: 8, 865: 8, 869: 7, 870: 7, 871: 2, 877: 8, 881: 8, 882: 8, 885: 8, 889: 8, 896: 8, 898: 8, 900: 6, 902: 6, 905: 8, 918: 8, 921: 8, 933: 8, 934: 8, 935: 8, 944: 8, 945: 8, 950: 8, 951: 8, 953: 8, 955: 8, 956: 8, 971: 7, 975: 5, 987: 8, 993: 8, 998: 5, 999: 7, 1000: 8, 1001: 8, 1002: 8, 1014: 8, 1017: 8, 1020: 8, 1041: 8, 1042: 8, 1044: 8, 1056: 8, 1057: 8, 1059: 1, 1063: 8, 1071: 8, 1076: 8, 1077: 8, 1082: 8, 1084: 8, 1085: 8, 1086: 8, 1114: 8, 1132: 8, 1161: 8, 1162: 8, 1163: 8, 1164: 8, 1165: 8, 1166: 8, 1167: 8, 1172: 8, 1228: 8, 1235: 8, 1237: 8, 1264: 8, 1279: 8, 1541: 8, 1552: 8, 1553: 8, 1556: 8, 1557: 8, 1568: 8, 1570: 8, 1571: 8, 1572: 8, 1592: 8, 1594: 8, 1595: 8, 1649: 8, 1745: 8, 1775: 8, 1779: 8, 1786: 8, 1787: 8, 1788: 8, 1789: 8, 1990: 8, 1998: 8

**Comment by illumiN8i** [comment]
Could fingerprint conflict ports be upstreamed without the fingerprint for now so forks don't have to be maintained with each release?

**Comment by rbiasini** [comment]
@illumiN8i , yes, I'm fine merging in with no fingeprint. Btw, Rerunning the fingerprint conflict checker script now (`selfdrive/test/test_fingerprints.py`) does not show conflicts anymore. Not sure why it showed red flags before. Are you saying that the CarmyH fingerprint had to be commented out though to have OP working on sideplugmike's car?

**Comment by illumiN8i** [comment]
@rbiasini Correct, sideplugmike's car didn't work until I commented out the CamryH fingerprints. @wocsor Can you rebase and comment out the RAV4H_TSS2 fingerprint?

**Comment by wocsor** [comment]
@illumiN8i has another port which does not have conflicts.

**Pull request closed without merging**